### PR TITLE
Fix/itemcompiler abstract

### PR DESCRIPTION
**Pull request by jbout**

### CODE REVIEW

File: models/classes/class.ItemCompiler.php [hunk 1]
```diff
@@ -27,16 +27,8 @@
  * @package taoItems
  
  */
-class taoItems_models_classes_ItemCompiler extends tao_models_classes_Compiler
+abstract class taoItems_models_classes_ItemCompiler extends tao_models_classes_Compiler
```
Comment (chilly86) [inline models/classes/class.ItemCompiler.php]: are you sure that this class is not instantiated anywhere?

File: models/classes/class.ItemCompiler.php [hunk 2]
```diff
@@ -27,16 +27,8 @@
...
 {
-    /**
-     * @throws common_exception_BadRequest
```
Comment (chilly86) [inline models/classes/class.ItemCompiler.php]: if you'd put @inheritdoc here nothing would change

File: models/classes/class.ItemCompiler.php (hunk 1 above)
Comment (jbout) [reply models/classes/class.ItemCompiler.php]: yes

File: models/classes/class.ItemCompiler.php (hunk 2 above)
Comment (jbout) [reply models/classes/class.ItemCompiler.php]: But changing the class to abstract is a cleaner solution than keeping it as a concrete class with a function that needs to be overridden to make it usable.

**Pull request merged**